server/src/infra/repositories/machine-learning.repository.ts

```diff
@@ -16,7 +16,13 @@ import { readFile } from 'fs/promises';
 export class MachineLearningRepository implements IMachineLearningRepository {
   private async post<T>(url: string, input: TextModelInput | VisionModelInput, config: ModelConfig): Promise<T> {
     const formData = await this.getFormData(input, config);
-    const res = await fetch(`${url}/predict`, { method: 'POST', body: formData });
+    const res = await fetch(`${url}/predict`, { method: 'POST', body: formData }).catch((error) => {
+      if (error['cause']['code'] === 'ENOTFOUND') {
```

Comment:
I think we should allow a wider range of error codes. Depending on the environment it's conceivable that DNS might return an IP, but nothing is running at that address. In general I think we can just ignore the particular error - any code here still means we cannot connect, after all.

Reply:
The other two main cases where the request can fail are when:

1. A model is being downloaded and the request times out because it took too long
2. The ML service can't download the model

Reply:
Perhaps a retry/back-off would help here? I regularly hit this with larger batches on a remote instance, which is likely due to bandwidth contention and/or timeouts.

Reply:
Since this is running in a queue worker, any failures should get retried automatically, so I don't think we also need retry logic inside the task itself.

Reply:
Except we don't actually use the functionality right now. We try / catch the error, log it, and move on currently.

Reply:
As far as I know we have the queue configured for 3 retries (https://github.com/immich-app/immich/blob/e47e25e671b1c53c33d71659debfe1def86c6839/server/src/infra/infra.config.ts#L37). Is there another setting somewhere that stops that?

Reply:
It only retries when the job handler throws an error, which never happens.

Reply:
https://github.com/immich-app/immich/blob/main/server/src/domain/job/job.service.ts#L138